materialize-sql/table_mapping.go

```diff
@@ -128,6 +128,7 @@ func ResolveTable(shape TableShape, dialect Dialect) (Table, error) {
 	}
 
 	for _, key := range shape.Keys {
+		key.IsPrimaryKey = true
```

Comment:
This was a bit unsettling, the projections for table keys were not being marked with `IsPrimaryKey`, I made it so, but I wonder if there was a reason they were not flagged as `IsPrimaryKey`? @willdonnelly

Reply:
I'm not aware of any reasons, though I'm also not super familiar with how that logic has evolved. Just from skimming the relevant code it looks like the `IsPrimaryKey` field of a `Column` is never actually used for anything -- the primary-keyness is implicit in the column being part of `shape.Keys` here. So it's probably just an oversight.